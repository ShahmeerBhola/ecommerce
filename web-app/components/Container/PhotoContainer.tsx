import React from 'react';
import Container from './Container';
import PhotoBackground from '../PhotoBackground';

type Props = {
    image?: string;
    className?: string;
    backgroundCover?: boolean;
    backgroundClassName?: string;
    backgroundStyles?: React.CSSProperties;
    containerPaddingY?: string;
};

const PhotoContainer: React.FunctionComponent<Props> = ({
    children,
    image,
    className = '',
    backgroundCover,
    backgroundClassName,
    backgroundStyles,
    containerPaddingY,
}) => {
    return (
        <PhotoBackground
            image={image ? `/images/${image}` : undefined}
            cover={backgroundCover}
            className={backgroundClassName}
            styles={backgroundStyles}
        >
            <Container className={className} paddingY={containerPaddingY}>
                {children}
            </Container>
        </PhotoBackground>
    );
};

export default PhotoContainer;
