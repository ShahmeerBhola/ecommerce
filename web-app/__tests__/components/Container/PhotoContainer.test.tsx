import React from 'react';
import renderer from 'react-test-renderer';
import PhotoContainer from '../../../components/Container/PhotoContainer';

describe('Container component', () => {
  test('it renders properly', () => {
    const tree = renderer
      .create(
        <PhotoContainer image="my-truck.png">
          <p>Hello</p>
        </PhotoContainer>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly with non-default props', () => {
    const tree = renderer
      .create(
        <PhotoContainer
          image="my-truck2.png"
          className="hello-there"
          backgroundCover={false}
          backgroundClassName="background-class-name"
          minHeight="50"
        >
          <p>Hello</p>
        </PhotoContainer>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when backgroundCover is set to true', () => {
    const tree = renderer
      .create(
        <PhotoContainer image="my-truck3.png" backgroundCover={true}>
          <p>Hello</p>
        </PhotoContainer>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
