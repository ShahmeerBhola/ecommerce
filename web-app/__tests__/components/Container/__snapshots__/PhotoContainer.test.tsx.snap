// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component it renders properly 1`] = `
<div
  className="bg-no-repeat bg-cover"
  style={
    Object {
      "backgroundImage": "url(/images/my-truck.png)",
    }
  }
>
  <div
    className="container mx-auto px-5 md:py-20 py-5 min-h-80"
  >
    <p>
      Hello
    </p>
  </div>
</div>
`;

exports[`Container component it renders properly when backgroundCover is set to true 1`] = `
<div
  className="bg-no-repeat bg-cover"
  style={
    Object {
      "backgroundImage": "url(/images/my-truck3.png)",
    }
  }
>
  <div
    className="container mx-auto px-5 md:py-20 py-5 min-h-80"
  >
    <p>
      Hello
    </p>
  </div>
</div>
`;

exports[`Container component it renders properly with non-default props 1`] = `
<div
  className="bg-no-repeat background-class-name"
  style={
    Object {
      "backgroundImage": "url(/images/my-truck2.png)",
    }
  }
>
  <div
    className="container mx-auto px-5 md:py-20 py-5 50 hello-there"
  >
    <p>
      Hello
    </p>
  </div>
</div>
`;
